apps/fulltextsearch.sh

```diff
@@ -87,51 +89,167 @@ fi
+  config_version: 2
+
+# Roles mapping
+all_access:
+  reserved: false
+  backend_roles:
+  - "admin"
+  description: "Maps admin to all_access"
+YML_ROLES_MAPPING
+
+# docker-compose.yml
+cat << YML_DOCKER_COMPOSE > $OPNSDIR/docker-compose.yml
+version: '3'
+services:
+  fts_os-node:
+    image: opensearchproject/opensearch
+    container_name: fts_os-node
+    command:
+      - sh
+      - -c
+      - "/usr/share/opensearch/bin/opensearch-plugin list | grep -q ingest-attachment \
+         || /usr/share/opensearch/bin/opensearch-plugin install --batch ingest-attachment ;
+         ./opensearch-docker-entrypoint.sh"
+    environment:
+      - cluster.name=fts_os-cluster
+      - node.name=fts_os-node
+      - bootstrap.memory_lock=true
+      - "OPENSEARCH_JAVA_OPTS=-Xms1024M -Xmx1024M"
+    ulimits:
+      memlock:
+        soft: -1
+        hard: -1
+      nofile:
+        soft: 65536
+        hard: 65536
+    volumes:
+      - fts_os-data:/usr/share/opensearch/data
+      - $OPNSDIR/root-ca.pem:/usr/share/opensearch/config/root-ca.pem
+      - $OPNSDIR/node.pem:/usr/share/opensearch/config/node.pem
+      - $OPNSDIR/node-key.pem:/usr/share/opensearch/config/node-key.pem
+      - $OPNSDIR/admin.pem:/usr/share/opensearch/config/admin.pem
+      - $OPNSDIR/admin-key.pem:/usr/share/opensearch/config/admin-key.pem
+      - $OPNSDIR/opensearch.yml:/usr/share/opensearch/config/opensearch.yml
+      - $OPNSDIR/internal_users.yml:/usr/share/opensearch/plugins/opensearch-security/securityconfig/internal_users.yml
+      - $OPNSDIR/roles_mapping.yml:/usr/share/opensearch/plugins/opensearch-security/securityconfig/roles_mapping.yml
+    ports:
+      - 127.0.0.1:9200:9200
+      - 127.0.0.1:9600:9600 # Performance Analyzer [1]
+    networks:
+      - fts_os-net
+
+volumes:
+  fts_os-data:
+
+networks:
+  fts_os-net:
+
+#[1] https://github.com/opensearch-project/performance-analyzer
+YML_DOCKER_COMPOSE
+
+# Prepare certs
+#create_certs opensearch_certs.sh
+rm "$OPNSDIR"/opensearch_certs.sh
+curl -s https://raw.githubusercontent.com/Ark74/vm/add_opensearch_fts_engine/static/opensearch_certs.sh > "$OPNSDIR"/opensearch_certs.sh
+sed -i "s|__NCDOMAIN__|$NCDOMAIN|" "$OPNSDIR"/opensearch_certs.sh
+cd "$OPNSDIR"
+bash opensearch_certs.sh
 
 # Set permissions
-chown 1000:1000 -R  $RORDIR
-chmod ug+rwx -R  $RORDIR
-
-# Run Elastic Search Docker
-docker run -d --restart always \
---name $fts_es_name \
---ulimit memlock=-1:-1 \
---ulimit nofile=65536:65536 \
--p 127.0.0.1:9200:9200 \
--p 127.0.0.1:9300:9300 \
--v esdata:/usr/share/elasticsearch/data \
--v /opt/es/readonlyrest.yml:/usr/share/elasticsearch/config/readonlyrest.yml \
--e "discovery.type=single-node" \
--e "bootstrap.memory_lock=true" \
--e ES_JAVA_OPTS="-Xms1024M -Xmx1024M -Dlog4j2.formatMsgNoLookups=true" \
--i -t $nc_fts
+chmod 744 -R  $OPNSDIR
+
+# Launch docker-compose
+cd $OPNSDIR
+docker-compose up -d
 
 # Wait for bootstrapping
-docker restart $fts_es_name
+docker restart $fts_node
```

Comment:
do we really need a restart here?

Reply:
not really, i just replace the old variable.

Reply:
okay, would be cool to get rid of that then :)